pkg/distributor/distributor.go

```diff
@@ -1846,6 +1846,45 @@ func (d *Distributor) ActiveSeries(ctx context.Context, matchers []*labels.Match
 	return deduplicatedSeries, nil
 }
 
+// activeSeriesResponse is a helper to merge/deduplicate ActiveSeries responses from ingesters.
+type activeSeriesResponse struct {
+	m       sync.Mutex
+	series  map[uint64]labels.Labels
+	builder labels.ScratchBuilder
+	lbls    labels.Labels
+}
+
+func newActiveSeriesResponse() *activeSeriesResponse {
+	return &activeSeriesResponse{
+		series:  make(map[uint64]labels.Labels),
+		builder: labels.NewScratchBuilder(40),
+	}
+}
+
+func (r *activeSeriesResponse) add(series []*mimirpb.Metric) {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	for _, metric := range series {
+		mimirpb.FromLabelAdaptersOverwriteLabels(&r.builder, metric.Labels, &r.lbls)
+		lblHash := r.lbls.Hash()
+		if _, ok := r.series[lblHash]; !ok {
+			r.series[lblHash] = r.lbls.Copy()
```

Comment:
it would be useful (for me) if there was a comment explaining why we need to copy

Reply:
I'm guessing the labels will get returned to some pool?

Reply:
oh, ic, because `r.lbls` gets overwritten again